### ISSUE TITLE
Fix file promotion and updated versioned FIX_DIR paths in workflow

### ISSUE THREAD

**Issue by KateFriedman-NOAA** [NOAA-EMC/global-workflow#966]
**Description**

This issue will document the promotion of `fix_NEW` to `fix` and the associated workflow linking updates to the newly restructured and versioned fix file sets on supported platforms. The initial promotion tasks will happen over a two day period and then the workflow updates will occur.

Checklist

August 10th (promotion day 1)
- [x] rename `fix` to `fix_20210923` (Hera)
- [x] rename `fix` to `fix_20210923` (WCOSS2 Cactus)
- [x] rename `fix` to `fix_20210923` (WCOSS2 Dogwood)
- [x] rename `fix` to `fix_20210923` (Orion)
- [x] rename `fix` to `fix_20210923` (Jet)
- [x] copy new `fix` in to replace `fix` (Hera)
- [x] rsync new `fix` from Hera in to replace `fix` on WCOSS2 Cactus
- [x] rsync new `fix` from Hera in to replace `fix` on WCOSS2 Dogwood
- [x] rsync new `fix` from Hera in to replace `fix` on Orion
- [x] rsync new `fix` from Hera in to replace `fix` on Jet
- [x] copy test version of `fix_NEW` (`fix_NEW_test`) into global space for sanity testing on Hera
- [x] perform sanity tests on `fix_NEW_test` on Hera

August 11th (promotion day 2)
- [x] rename `fix_NEW` to `fix_NEW_20220726` (Hera)
- [x] rename `fix_NEW` to `fix_NEW_20220726` (WCOSS2 Cactus)
- [x] rename `fix_NEW` to `fix_NEW_20220726` (WCOSS2 Dogwood)
- [x] rename `fix_NEW` to `fix_NEW_20220726` (Orion)
- [x] rename `fix_NEW` to `fix_NEW_20220726` (Jet)
- [x] create new `fix_NEW` with symlinks to updated `fix` (Hera)
- [x] create new `fix_NEW` with symlinks to updated `fix` (WCOSS2 Cactus)
- [x] create new `fix_NEW` with symlinks to updated `fix` (WCOSS2 Dogwood)
- [x] create new `fix_NEW` with symlinks to updated `fix` (Orion)
- [x] create new `fix_NEW` with symlinks to updated `fix` (Jet)

August 19th (cleanup)
- [x] delete `fix_NEW_20220726` (Hera)
- [x] delete `fix_NEW_20220726` (WCOSS2 Cactus)
- [x] delete `fix_NEW_20220726` (WCOSS2 Dogwood)
- [x] delete `fix_NEW_20220726` (Orion) 
- [x] delete `fix_NEW_20220726` (Jet)

August 26th (cleanup)
- [x] delete `fix_20210923` (Hera)
- [x] delete `fix_20210923` (WCOSS2 Cactus)
- [x] delete `fix_20210923` (WCOSS2 Dogwood)
- [x] delete `fix_20210923` (Orion) 
- [x] delete `fix_20210923` (Jet)

Workflow updates
- [x] update `link_workflow.sh` `FIX_DIR`s to point to new `fix` (https://github.com/NOAA-EMC/global-workflow/pull/1009)
- [x] introduce fix file versioning for `FIX_DIR` paths in `link_workflow.sh` (https://github.com/NOAA-EMC/global-workflow/pull/1009)
- [x] update scripts based on fix subfolder renaming (https://github.com/NOAA-EMC/global-workflow/pull/1009)
- [x] dependency: update UFS_UTILS version (https://github.com/ufs-community/UFS_UTILS/issues/687)

**Comment by KateFriedman-NOAA** [comment]
Ran control (with fix_NEW) and test (using updated fix paths) on Orion (both atmos-only and coupled). The test output matches the control run.

atmos-only run spot checking:
```
Orion-login-1[39] /work/noaa/stmp/kfriedma/comrot/fixveratmos$ cmp ../devctrlatmos/gdas.20220101/18/atmos/gdas.t18z.master.grb2f009 gdas.20220101/18/atmos/gdas.t18z.master.grb2f009
Orion-login-1[40] /work/noaa/stmp/kfriedma/comrot/fixveratmos$ cmp ../devctrlatmos/gdas.20220101/18/atmos/gdas.t18z.sfluxgrbf009.grib2 gdas.20220101/18/atmos/gdas.t18z.sfluxgrbf009.grib2
Orion-login-1[42] /work/noaa/stmp/kfriedma/comrot/fixveratmos$ cmp ../devctrlatmos/gdas.20220102/00/atmos/gdas.t00z.master.grb2f009 gdas.20220102/00/atmos/gdas.t00z.master.grb2f009
Orion-login-1[43] /work/noaa/stmp/kfriedma/comrot/fixveratmos$ cmp ../devctrlatmos/gdas.20220102/00/atmos/gdas.t00z.sfluxgrbf009.grib2 gdas.20220102/00/atmos/gdas.t00z.sfluxgrbf009.grib2
Orion-login-1[50] /work/noaa/stmp/kfriedma/comrot/fixveratmos$ nccmp -dgB ../devctrlatmos/gdas.20220102/00/atmos/gdas.t00z.sfcf009.nc gdas.20220102/00/atmos/gdas.t00z.sfcf009.nc
Orion-login-1[52] /work/noaa/stmp/kfriedma/comrot/fixveratmos$ nccmp -dgB ../devctrlatmos/gfs.20220102/00/atmos/gfs.t00z.sfcf024.nc gfs.20220102/00/atmos/gfs.t00z.sfcf024.nc
Orion-login-1[53] /work/noaa/stmp/kfriedma/comrot/fixveratmos$ nccmp -dgB ../devctrlatmos/gfs.20220102/00/atmos/gfs.t00z.atmf024.nc gfs.20220102/00/atmos/gfs.t00z.atmf024.nc
Orion-login-1[54] /work/noaa/stmp/kfriedma/comrot/fixveratmos$
```

**Comment by KateFriedman-NOAA** [comment]
Ran `test/diff_ROTDIR.sh` on coupled control and test using updated fix file set. Everything came back as identical/matching between the runs.

**Comment by KateFriedman-NOAA** [comment]
@GeorgeGayno-NOAA Below are the UFS_UTILS files I had to modify for the updated fix subfolders (see clone on Orion):

```
Orion-login-1[151] /work/noaa/global/kfriedma/git/feature-fix-versioning/sorc/ufs_utils.fd$ git log -1
commit a2b0817028e91aca4233c2d4198d14049bd6abbd
Author: GeorgeGayno-NOAA <52789452+GeorgeGayno-NOAA@users.noreply.github.com>
Date:   Wed Jun 1 14:34:01 2022 -0400

    Update documentation for SRW App (#656)

    Update the doxygen and 'readthedocs' documentation. Update the
    varmap file 'readme' file. Supports v2.0 of the SRW Application.

    Fixes #643.
Orion-login-1[150] /work/noaa/global/kfriedma/git/feature-fix-versioning/sorc/ufs_utils.fd$ git status
# HEAD detached at a2b0817
# Changes not staged for commit:
#   (use "git add <file>..." to update what will be committed)
#   (use "git checkout -- <file>..." to discard changes in working directory)
#
#       modified:   fix/link_fixdirs.sh
#       modified:   scripts/exemcsfc_global_sfc_prep.sh
#       modified:   ush/emcsfc_ice_blend.sh
#       modified:   ush/emcsfc_snow.sh
#       modified:   ush/fv3gfs_driver_grid.sh
#       modified:   ush/global_cycle.sh
#       modified:   ush/global_cycle_driver.sh
#
# Untracked files:
#   (use "git add <file>..." to include in what will be committed)
#
#       fix/am
#       fix/orog
#       fix/sfc_climo
no changes added to commit (use "git add" and/or "git commit -a")
```

We'll need you to make these same changes in UFS_UTILS develop and then we'll update the UFS_UTILS version as the related global-workflow changes go in. I have another issue (#974) that competes with this work a bit. May need to coordinate when we commit these fix changes for after we update to the newer UFS_UTILS tag. Not 100% sure at the moment.

**Comment by GeorgeGayno-NOAA** [comment]
> @GeorgeGayno-NOAA Below are the UFS_UTILS files I had to modify for the updated fix subfolders (see clone on Orion):
> 
> ```
> Orion-login-1[151] /work/noaa/global/kfriedma/git/feature-fix-versioning/sorc/ufs_utils.fd$ git log -1
> commit a2b0817028e91aca4233c2d4198d14049bd6abbd
> Author: GeorgeGayno-NOAA <52789452+GeorgeGayno-NOAA@users.noreply.github.com>
> Date:   Wed Jun 1 14:34:01 2022 -0400
> 
>     Update documentation for SRW App (#656)
> 
>     Update the doxygen and 'readthedocs' documentation. Update the
>     varmap file 'readme' file. Supports v2.0 of the SRW Application.
> 
>     Fixes #643.
> Orion-login-1[150] /work/noaa/global/kfriedma/git/feature-fix-versioning/sorc/ufs_utils.fd$ git status
> # HEAD detached at a2b0817
> # Changes not staged for commit:
> #   (use "git add <file>..." to update what will be committed)
> #   (use "git checkout -- <file>..." to discard changes in working directory)
> #
> #       modified:   fix/link_fixdirs.sh
> #       modified:   scripts/exemcsfc_global_sfc_prep.sh
> #       modified:   ush/emcsfc_ice_blend.sh
> #       modified:   ush/emcsfc_snow.sh
> #       modified:   ush/fv3gfs_driver_grid.sh
> #       modified:   ush/global_cycle.sh
> #       modified:   ush/global_cycle_driver.sh
> #
> # Untracked files:
> #   (use "git add <file>..." to include in what will be committed)
> #
> #       fix/am
> #       fix/orog
> #       fix/sfc_climo
> no changes added to commit (use "git add" and/or "git commit -a")
> ```
> 
> We'll need you to make these same changes in UFS_UTILS develop and then we'll update the UFS_UTILS version as the related global-workflow changes go in. I have another issue (#974) that competes with this work a bit. May need to coordinate when we commit these fix changes for after we update to the newer UFS_UTILS tag. Not 100% sure at the moment.

Just opened an issue to track this: https://github.com/ufs-community/UFS_UTILS/issues/687

**Comment by KateFriedman-NOAA** [comment]
> Just opened an issue to track this: https://github.com/ufs-community/UFS_UTILS/issues/687

Awesome, thanks @GeorgeGayno-NOAA !

**Comment by bubba0077** [comment]
Just a note here to make sure we're on the same page: link_fixdirs should probably take in the versions just like we do for modulefiles. Not sure if Kate's current changes do that, but the final solution should.

**Comment by WalterKolczynski-NOAA** [comment]
Just a note here to make sure we're on the same page: link_fixdirs should probably take in the versions just like we do for modulefiles. Not sure if Kate's current changes do that, but the final solution should.

**Comment by KateFriedman-NOAA** [comment]
Great point @WalterKolczynski-NOAA , thanks!

@GeorgeGayno-NOAA Please see the link script changes I made in my g-w branch here: https://github.com/NOAA-EMC/global-workflow/commit/fdd640fdce71287fcf3c19b96dc94da0f9682c22

I also created a `fix.ver` file to hold the subfolder versions, which `link_workflow.sh` uses when creating the symlinks under fix:
https://github.com/NOAA-EMC/global-workflow/blob/feature/fix-versioning/versions/fix.ver

I updated my changes to my UFS_UTILS copy to show what the symlinks should now be based on the contents of `fix.ver`:
```
Orion-login-1[19] /work/noaa/global/kfriedma/git/feature-fix-versioning/sorc/ufs_utils.fd/fix$ pwd
/work/noaa/global/kfriedma/git/feature-fix-versioning/sorc/ufs_utils.fd/fix
Orion-login-1[20] /work/noaa/global/kfriedma/git/feature-fix-versioning/sorc/ufs_utils.fd/fix$ ll
total 4
lrwxrwxrwx 1 kfriedma global   41 Aug 25 16:03 am -> /work/noaa/global/glopara/fix/am/20220805/
-rwxr-xr-x 1 kfriedma global 1699 Aug 25 14:33 link_fixdirs.sh*
lrwxrwxrwx 1 kfriedma global   43 Aug 25 16:04 orog -> /work/noaa/global/glopara/fix/orog/20220805/
lrwxrwxrwx 1 kfriedma global   48 Aug 25 16:04 sfc_climo -> /work/noaa/global/glopara/fix/sfc_climo/20220805/
```

**Comment by KateFriedman-NOAA** [comment]
@WalterKolczynski-NOAA fyi, the following changes will need to be made to the GLDAS scripts for this issue:

```
diff --git a/jobs/JGDAS_ATMOS_GLDAS b/jobs/JGDAS_ATMOS_GLDAS
index a75e814..caa1684 100755
--- a/jobs/JGDAS_ATMOS_GLDAS
+++ b/jobs/JGDAS_ATMOS_GLDAS
@@ -74,7 +74,7 @@ export COMPONENT=${COMPONENT:-atmos}
 ##############################################
 export gldas_ver=${gldas_ver:-v2.3.0}
 export HOMEgldas=${HOMEgldas:-$HOMEgfs}
-export FIXgldas=${FIXgldas:-$HOMEgldas/fix/fix_gldas}
+export FIXgldas=${FIXgldas:-$HOMEgldas/fix/gldas}
 export PARMgldas=${PARMgldas:-$HOMEgldas/parm/gldas}
 export EXECgldas=${EXECgldas:-$HOMEgldas/exec}
 export USHgldas=${USHgldas:-$HOMEgldas/ush}

diff --git a/scripts/exgdas_atmos_gldas.sh b/scripts/exgdas_atmos_gldas.sh
index ad14fc6..7697712 100755
--- a/scripts/exgdas_atmos_gldas.sh
+++ b/scripts/exgdas_atmos_gldas.sh
@@ -61,7 +61,7 @@ export nlat=$((2*res))
 export nlon=$((4*res))

 export FIXgldas=${FIXgldas:-$HOMEgfs/fix}
-export topodir=${topodir:-$HOMEgfs/fix/fix_fv3_gmted2010/$CASE/}
+export topodir=${topodir:-$HOMEgfs/fix/orog/$CASE/}

```
